src/dbnode/storage/shard.go

```diff
@@ -392,7 +392,20 @@ func (s *dbShard) Stream(
 	onRetrieve block.OnRetrieveBlock,
 	nsCtx namespace.Context,
 ) (xio.BlockReader, error) {
-	return s.DatabaseBlockRetriever.Stream(ctx, s.shard, id, blockStart, onRetrieve, nsCtx)
+	return s.DatabaseBlockRetriever.Stream(ctx, s.shard, id,
+		blockStart, onRetrieve, nsCtx)
+}
+
+// StreamIndexChecksum implements series.QueryableBlockRetriever
+func (s *dbShard) StreamIndexChecksum(
```

Comment:
For `Stream` method I would expect multiple results to be streamed, but from what I can tell this only gives one. I suggest renaming to `FetchIndexChecksumAsync` or smth. similar.

Reply:
I'd prefer to keep the current naming to mirror the existing Stream method, since it performs in a similar manner (that method also returns only the data for a single series ID within the block rather than multiple series, for reference)

Reply:
+1